File: newtonRaphson/newton.ml
(* Newton Raphson method *)

let eps = 0.001 in

let rec newton_raphson x a =  
   let improv x = (x +. a/.x)/.2. in 

   let satis x = 
      if (abs_float(x*.x -. a) < eps) then 
         true
      else 
         false
   in

   if satis x then
      let () = Printf.printf "Sq %f = %f\n" a x in
      x
   else 
      newton_raphson (improv x) a 
in

(* Assuming initial guess to be the same as the number *)
let square_root x = 
   newton_raphson x x
in

square_root 5.0;;

File: newtonRaphson/cpp/newton.cpp
#include "newton.h"
#include <math.h>
#include <iostream>

const double Newton::EPS = 0.001; 

const double Newton::squareRoot(const double x)
{
   double y = x; // Initial guess
   while (fabs(pow(y,2) - x)  > EPS)
   {
      std::cout << " new val" << y << std::endl;
      y = (y + x/y)/2.0;
   }
   return y; 
}
